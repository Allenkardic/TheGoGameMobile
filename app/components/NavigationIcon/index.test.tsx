// import React from 'react';
// import renderer from 'react-test-renderer';
// import {Pressable} from 'react-native';
// import NavigationIcon from '.'; // Adjust the path to your component
// import Icon from 'react-native-vector-icons/Feather'; // Ensure Icon is properly imported

// describe('NavigationIcon Component', () => {
//   it('renders correctly with default icon', () => {
//     const component = renderer.create(<NavigationIcon onPress={() => {}} />);
//     const tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   it('renders with custom icon name', () => {
//     const component = renderer.create(
//       <NavigationIcon onPress={() => {}} name="home" />,
//     );
//     const root = component.root;
//     const icon = root.findByType(Icon);

//     expect(icon.props.name).toBe('home');
//   });

//   it('renders the correct default icon when no name is passed', () => {
//     const component = renderer.create(<NavigationIcon onPress={() => {}} />);
//     const root = component.root;
//     const icon = root.findByType(Icon); // Ensure the Icon is rendered

//     expect(icon.props.name).toBe('arrow-left'); // Default name is 'arrow-left'
//   });

//   it('handles onPress correctly', () => {
//     const mockOnPress = jest.fn();
//     const component = renderer.create(<NavigationIcon onPress={mockOnPress} />);
//     const root = component.root;
//     const pressable = root.findByType(Pressable);

//     pressable.props.onPress(); // Simulate the press action

//     expect(mockOnPress).toHaveBeenCalled();
//   });
// });
