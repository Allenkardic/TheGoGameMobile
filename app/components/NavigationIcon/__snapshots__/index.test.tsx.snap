// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationIcon Component renders correctly with default icon 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="navigation-icon"
>
  <Icon
    name="arrow-left"
    size={24}
  />
</View>
`;
